src/relstorage/component.xml

```diff
@@ -211,6 +211,12 @@
       </description>
     </key>
 
+    <key name="ssl" datatype="boolean" required="no">
+      <description>
+        if set, connection uses ssl
+      </description>
+    </key>
+
```

Comment:
Unfortunately, this doesn't work. Fortunately, it seems to not be necessary because connectors default to SSL by default anyway.

Both [MySQLdb](https://github.com/PyMySQL/mysqlclient/blob/2316313fc6777591494e870ba0c5da0d3dcd6dc6/MySQLdb/connections.py#L124) and [PyMySQL](https://github.com/PyMySQL/PyMySQL/blob/f24cb9aa7295921bcd8f34f752c8a05b981d3125/pymysql/connections.py#L129) define the `ssl` argument to be a *dictionary* containing a number of different parameters. 


In PyMySQL, if `ssl` isn't a dictionary, you get a stack trace:
```python traceback
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "///ZODB/config.py", line 94, in storageFromFile
    return storageFromConfig(config.storage)
  File "///ZODB/config.py", line 104, in storageFromConfig
    return section.open()
  File "///zc/zlibstorage/__init__.py", line 211, in open
    base = self.config.base.open()
  File "//src/relstorage/config.py", line 43, in open
    return RelStorage(adapter, name=config.name, options=options)
  File "//src/relstorage/storage/__init__.py", line 174, in __init__
    self._adapter.schema.prepare()
  File "//src/relstorage/adapters/schema.py", line 738, in prepare
    self.connmanager.open_and_call(self._prepare_with_connection)
  File "//src/relstorage/adapters/connmanager.py", line 290, in open_and_call
    conn, cursor = self._do_open_for_call(callback)
  File "//src/relstorage/adapters/connmanager.py", line 364, in _do_open_for_call
    return self.open(**_connection_callback_open_args(self, callback))
  File "//src/relstorage/adapters/mysql/connmanager.py", line 103, in open
    conn = self._db_connect(**params)
  File "//src/relstorage/adapters/mysql/drivers/pymysql.py", line 87, in connect
    return AbstractMySQLDriver.connect(self, *args, **kwargs)
  File "//src/relstorage/adapters/mysql/drivers/__init__.py", line 165, in connect
    return super(AbstractMySQLDriver, self).connect(*args, **kwargs)
  File "//src/relstorage/adapters/drivers.py", line 173, in connect
    return self._connect(*args, **kwargs)
  File "///pymysql/connections.py", line 285, in __init__
    self.ctx = self._create_ssl_ctx(ssl)
  File "///pymysql/connections.py", line 365, in _create_ssl_ctx
    ca = sslp.get("ca")
AttributeError: 'bool' object has no attribute 'get'
```

With MySQLdb, if the `ssl` argument isn't a dictionary, or doesn't contain the right parameters, `MySQLdb` [silently ignores the errors](https://github.com/PyMySQL/mysqlclient/blob/2316313fc6777591494e870ba0c5da0d3dcd6dc6/MySQLdb/_mysql.c#L458-L468) and calls [mysql_ssl_set](https://dev.mysql.com/doc/c-api/5.7/en/mysql-ssl-set.html) with a bunch of `NULL` values. This simply has the effect of using the defaults, which ultimately does nothing [because](https://dev.mysql.com/doc/c-api/5.7/en/mysql-ssl-set.html): 

> It is optional to call `mysql_ssl_set()` to obtain an encrypted connection because by default, MySQL programs attempt to connect using encryption if the server supports encrypted connections

Further, even if you call this with some set of parameters, you're not guaranteed to get an encrypted connection:

> `mysql_ssl_set()` specifies SSL information such as certificate and key files for establishing an encrypted connection if such connections are available,  **does not enforce any requirement that the connection obtained be encrypted.** 

Finally, `MySQL Connector` is natuarlly the oddball out. It [requires a separate parameter](https://dev.mysql.com/doc/connector-python/en/connector-python-connectargs.html) for each aspect of SSL: `ssl_ca`, `ssl_key`, etc, and produces `AttributeError: Unsupported argument 'ssl'` if you attempt to pass an `ssl` argument. (PyMySQL can also take individual arguments.) However, it also defaults to using SSL if available, though.

If configuring SSL is desired, it would need to:

1. Allow configuring the SSL settings like CA, Certificate, Key, etc.
2. Work with all driver implementations
3. Where the driver supports it, [enforce that SSL is actually being used](https://dev.mysql.com/doc/c-api/5.7/en/c-api-encrypted-connections.html#c-api-enforcing-encrypted-connection). For MySQLdb, this is the `ssl_mode` argument; I'm not sure about the other drivers.
3. Have test cases to ensure that this is doing the right thing.